internal/processor/filtermetric/config.go

```diff
@@ -16,13 +16,31 @@ package filtermetric
 
 import (
 	"go.opentelemetry.io/collector/internal/processor/filterset"
+	"go.opentelemetry.io/collector/internal/processor/filterset/regexp"
+)
+
+// MatchType specifies the strategy for matching against `pdata.Metric`s. This
+// is distinct from filterset.MatchType which matches against metric (and
+// tracing) names only. To support matching against metric names and
+// `pdata.Metric`s, filtermetric.MatchType is effectively a superset of
+// filterset.MatchType.
+type MatchType string
```

Comment:
Is there an advantage of redeclaring this type and not using from filterset directly?

Reply:
We could put the `expr` MatchType in the `filterset` package but I don't think it belongs there because the `filterset` package and the `filterset.MatchType`s (`strict` and `regexp`) are just for string matching (and not just for metric names, they're also used to match traces).

The `expr` MatchType OTOH is not for matching just metric names, but for matching `pdata.Metric`s, which I think means it should live here in the `filtermetric` package. In addition, because you can support both metric name matching and full metric matching if you're given a `pdata.Metric`, this package will be used (in the next PR) to hold the types that will perform both kinds of matching. So, in effect, the `MatchType`s in this package are a superset of those in `filterset`. 

I think most of the wonkiness is because, for ease of use, we're sort of overloading the meaning of `match_type` in the config. On the one hand, it can be `regexp` or `strict` for metric name matching, or on the other hand, `expr` for full metric matching.

Having said all that, I'm very much open to suggestions if there's a way to make this cleaner.

Reply:
Thanks for the clarification.  I was referring to general type reuse w/o larger semantic implications, but can understand wanting to keep things separate.

```
const (
	Regexp = filterset.Regexp
	Strict = filterset.Strict
	Expr filterset.MatchType = "expr"
)
```

Reply:
Oh sorry, yeah, I fixed this in the next PR. :-)

Reply:
Fixed in this one too.